File: taskflowbackend/src/main/java/com/taskflow/taskflowbackend/service/TaskService.java
package com.taskflow.taskflowbackend.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.taskflow.taskflowbackend.entity.Task;
import com.taskflow.taskflowbackend.repository.TaskRepository;

@Service

public class TaskService {

    @Autowired

    private TaskRepository taskRepository;

    // CREATE TASK

    public Task createTask(Task task) {

        return taskRepository.save(task);

    }

    // GET ALL TASKS

    public List<Task> getAllTasks() {

        return taskRepository.findAll();

    }

    // DELETE TASK

    public void deleteTask(Long id) {

        taskRepository.deleteById(id);

    }

}

File: frontend/src/services/taskService.js
import API from "./api";

// CREATE TASK

export const createTask = async (taskData) => {

    const response = await API.post("/tasks", taskData);

    return response.data;
};

// GET TASKS

export const getTasks = async () => {

    const response = await API.get("/tasks");

    return response.data;
};

// GET ALL TASKS

export const getAllTasks = async () => {

    const response = await API.get("/tasks");

    return response.data;
};

// DELETE TASK

export const deleteTask = async (id) => {

    const response = await API.delete(`/tasks/${id}`);

    return response.data;
};